File: scripts/release/context_sdk_consumer.py
"""Installed-wheel/sdist cross-language oracle consumer. No source-tree SDK imports."""

import json
import sys
from pathlib import Path

import cigar_sdk
from cigar_sdk import LocalContextError, LocalContextGraph

cases = json.loads(Path(sys.argv[1]).read_bytes())
worker = sys.argv[2] if len(sys.argv) > 2 else None
results = []
for case in cases:
    try:
        with LocalContextGraph(case["domain"], worker_path=worker) as graph:
            for document in case["documents"]:
                graph.upsert(document)
            for from_id, to_id, kind in case["edges"]:
                graph.link(from_id, to_id, kind)
            result = graph.compile(case["request"])
            assert graph.verify(result["snapshot"]) == result
            results.append(result)
    except LocalContextError as error:
        results.append({"error": error.code})
print(
    json.dumps(
        {
            "exports": cigar_sdk.__all__,
            "module": cigar_sdk.__file__,
            "results": results,
        },
        ensure_ascii=False,
    )
)

File: scripts/release/context-sdk-consumer.mjs
import assert from "node:assert/strict";
import { readFileSync } from "node:fs";
import * as sdk from "@hol-org/cigar";
import { LocalContextGraph, LocalContextError } from "@hol-org/cigar/context";

const cases = JSON.parse(readFileSync(process.argv[2], "utf8"));
const results = [];
for (const item of cases) {
  try {
    await using graph = await LocalContextGraph.create(item.domain);
    for (const document of item.documents) await graph.upsert(document);
    for (const [from, to, kind] of item.edges) await graph.link(from, to, kind);
    const result = await graph.compile(item.request);
    assert.deepEqual(await graph.verify(result.snapshot), result);
    results.push(result);
  } catch (error) {
    if (!(error instanceof LocalContextError)) throw error;
    results.push({error: error.code});
  }
}
console.log(JSON.stringify({exports: Object.keys(sdk), results}));
